### DISCUSSION TITLE
Apply Filters During Queue

### DISCUSSION THREAD

**Discussion by andrewskm** [spatie/laravel-query-builder#777] [Q&A] [answered]
Hi,

I am using the query builder for users to filter and sort tabular data tables and it works perfectly.
One feature I am implementing is allowing the user to export the list (using Laravel Excel).
The export process needs to be queued (in case of large datasets), but during queue, the filters are lost.

Is there a standard way to allow the query builder to reapply these filters or do I need to go the route of extending the class **QueryBuilderRequest**?

**Comment by andrewskm** [comment, marked as answer]
Ended up solving this by realizing that you can send a request as the 2nd parameter to the `for()` method.
Placing what I needed to add just in case someone else is doing the same or similar.

In the excel export (using FromQuery):
``` php
public function __construct(public $requestQuery = null)
{
    $this->requestQuery = request()->query();
}

public function query()
{
    $request = app(QueryBuilderRequest::class)->merge($this->requestQuery);
    return QueryBuilder::for(User::class, $request);
}
```

**Reply by siarheipashkevich** [reply]
What about data from post request?

**Reply by andrewskm** [reply]
Should be able to do the same, but with `request()->all()` instead of `->query()` (that was my use case).

**Comment by camohub** [comment]
It does not work for me.
I did the same like 
```
        $request = app(QueryBuilderRequest::class)->merge(['filter' => $this->filter]);

        Log::info('$request->get(filter)');
        Log::info($request->get('filter'));  // its id=10

        return QueryBuilder::for(Customer::class, $request)
            ->where('id', '<', 20)
            ->toBase()
            ->orderByDesc('id');
```
I can see the $request->get('filter') is right but the QueryBuilder do not care about it. It does not filter the date according this filter.

**Comment by camohub** [comment]
I forgot to add allowedFilters() to the QueryBuilder
